server/src/domain/sql_migrations.rs

```diff
@@ -1131,6 +1140,91 @@ async fn migrate_to_v10(transaction: DatabaseTransaction) -> Result<DatabaseTran
+                .col(
+                    ColumnDef::new(UserAttributesSearch::UserAttributeSearchUserId)
+                        .string_len(255)
+                        .not_null(),
+                )
+                .col(
+                    ColumnDef::new(UserAttributesSearch::UserAttributeSearchName)
+                        .string_len(64)
+                        .not_null(),
+                )
+                .col(
+                    ColumnDef::new(UserAttributesSearch::UserAttributeSearchNo)
+                        .integer()
+                        .not_null(),
+                )
+                .col(
+                    ColumnDef::new(UserAttributesSearch::UserAttributeSearchText)
+                        .text()
+                        .not_null(),
+                )
+                .foreign_key(
+                    ForeignKey::create()
+                        .name("UserAttributeSearchUserIdForeignKey")
+                        .from(UserAttributesSearch::Table, UserAttributesSearch::UserAttributeSearchUserId)
+                        .to(Users::Table, Users::UserId)
+                        .on_delete(ForeignKeyAction::Cascade)
+                        .on_update(ForeignKeyAction::Cascade),
+                )
+                .foreign_key(
+                    ForeignKey::create()
+                        .name("UserAttributeSearchNameForeignKey")
+                        .from(UserAttributesSearch::Table, UserAttributesSearch::UserAttributeSearchName)
+                        .to(
+                            UserAttributeSchema::Table,
+                            UserAttributeSchema::UserAttributeSchemaName,
+                        )
+                        .on_delete(ForeignKeyAction::Cascade)
+                        .on_update(ForeignKeyAction::Cascade),
+                )
+                .primary_key(
+                    Index::create()
+                        .col(UserAttributesSearch::UserAttributeSearchUserId)
+                        .col(UserAttributesSearch::UserAttributeSearchName)
+                        .col(UserAttributesSearch::UserAttributeSearchNo),
+                ),
+        ),
+    )
+    .await?;
+
+    transaction
+            .execute(
+                builder.build(
+                    Query::insert()
+                        .into_table(UserAttributeSchema::Table)
+                        .columns([
+                            UserAttributeSchema::UserAttributeSchemaName,
+                            UserAttributeSchema::UserAttributeSchemaType,
+                            UserAttributeSchema::UserAttributeSchemaIsList,
+                            UserAttributeSchema::UserAttributeSchemaIsUserVisible,
+                            UserAttributeSchema::UserAttributeSchemaIsUserEditable,
+                            UserAttributeSchema::UserAttributeSchemaIsHardcoded,
+                        ])
+                        .values_panic([
+                            "mail".into(),
```

Comment:
Why are you adding the "mail" attribute there?

So far, I didn't convert it to a "custom" attribute because of the special requirements (it's used for password recovery, it needs to be non-empty, and up to now custom attributes were not searchable by substring, though you seem to have solved that one)

Reply:
I was getting foreign keys violation error because of inserting mail into this table https://github.com/lldap/lldap/pull/1000/files/8e2d89f2330fd7b89b3a0af81b3b3d12f3bfb244#diff-218546186ebdece3cd3f337cf80711f352fb77467f6e97c0cce99032e01aab47R466